src/Sylius/Component/Product/Generator/SlugGenerator.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Product\Generator;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class SlugGenerator implements SlugGeneratorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function generate($name)
+    {
+        return preg_replace('/[^a-z0-9\-]/', '', str_replace(' ', '-', strtolower($name)));
```

Comment:
We definitely need something better here, this won't support transliteration. Have a look at gedmo's code.

Reply:
@pjedrzejewski could you please explain the reason to use slug transliteration when the name translation is available and browsers supports UTF8 URLs?

Reply:
I have no clue what what was my thinking there, I think it was a pretty standard approach back then. Also, the assumption was that you can implement your own SlugGenerator and do whatever you prefer. :)

Reply:
@pjedrzejewski thanks for a quick answer. Agree, old SlugGenerator can be easily backported. Still, IMO Sylius should not transliterate because it might affect SEO. Not sure that Google Crawler process transliteration properly.

Reply:
@dbalabka I believe it does. I often write in Russian transliteration (because I'm not used to type in Cyrillic and I'm slow) and it works, so I don't see why it wouldn't work the other way around.
Will search for an indexed URL to test it on.